src/NexusMods.DataModel/ArchiveManager.cs

```diff
@@ -92,6 +92,20 @@ public bool HaveArchive(Hash hash)
         var rel = NameForHash(hash);
         return _locations.Any(r => r.CombineUnchecked(rel).FileExists);
     }
+    
+    /// <summary>
+    /// Determines if a given archive is stored by the manager, and returns the path if it is.
+    /// </summary>
+    /// <param name="hash"></param>
+    /// <param name="path"></param>
+    /// <returns></returns>
+    public bool TryGetPathFor(Hash hash, out AbsolutePath path)
+    {
+        var rel = NameForHash(hash);
+        path = _locations.Select(r => r.CombineUnchecked(rel))
+            .FirstOrDefault(r => r.FileExists);
+        return path != null;
+    }
```

Comment:
This was an oversight on my end while reviewing the code. `.FirstOrDefault` is notorious for breaking code when the item type of the collection is a value type, because a value type will return `default` instead of `null`. This means that the check `path != null` will **always** be true.

In #263 I added the enumerable extension method `TryGetFirst`, which should be used instead of `FirstOrDefault` for enumerable of collections with value type items. I'll fix this code in my upcoming PR for #266.